### PR TITLE
Add Kotlin DSL samples to upgrading_version_4 chapter

### PR DESCRIPTION
**Pull request by eskatos**

### CODE REVIEW

File: subprojects/docs/src/docs/userguide/upgrading_version_4.adoc [hunk 1]
```diff
@@ -634,13 +711,34 @@ As part of the stabilization effort in Gradle 4.8, we are deprecating this behav
...
 The new, stable behavior can be switched on by adding the following to your settings file:
 
-    enableFeaturePreview('STABLE_PUBLISHING')
+====
+[.multi-language-sample]
+=====
+.settings.gradle
+[source, groovy]
+----
+enableFeaturePreview('STABLE_PUBLISHING')
```
Comment (eriwen) [inline subprojects/docs/src/docs/userguide/upgrading_version_4.adoc]: IMO we should always suggest `"` in Groovy, but we already had that discussion and I will just let this go.

**Comment by eriwen** [comment]
Ship it @eskatos!

File: subprojects/docs/src/docs/userguide/upgrading_version_4.adoc [hunk 2]
```diff
@@ -469,9 +473,31 @@ task myOtherTask {
     doLast { ... }
 }
 ----
+=====
+[.multi-language-sample]
+=====
+.build.gradle.kts
+[source,kotlin]
+----
+class MyTask : DefaultTask() {
```
Comment (jmfayard) [inline subprojects/docs/src/docs/userguide/upgrading_version_4.adoc]: Didn't you forgot `open class` here and in the rest of the PR`

https://github.com/gradle/kotlin-dsl/pull/1240

Comment (eskatos) [reply subprojects/docs/src/docs/userguide/upgrading_version_4.adoc]: 👍 fixed

**Pull request merged**